multianalyzer/app/rebin.py

```diff
@@ -124,22 +127,42 @@ def parse():
     if options.debug:
         logger.setLevel(logging.DEBUG)
         logging.root.setLevel(level=logging.DEBUG)
+
     return options
 
 
-def rebin(options):
-    start_time = get_isotime()
-    t_start = time.perf_counter()
-    rt_read = 0.0
-    rt_rebin = 0.0
-    rt_write = 0.0
-    
-    print(f"Load topas refinement file: {options.pars}")
-    param = topas_parser(options.pars)
+def rebin_result_generator(infile=None, entry=None, hdf5_data=None, output=None, timer=None, pars=None, device=None, debug=None, energy=None, wavelength=None,
+               pixel=None, step=None, range=None, phi=None, width=None, delta2theta=None, iter=None, startp=None, endp=None):
```

Comment:
I don't understand the rational for unpacking options ...

Reply:
https://gitlab.esrf.fr/workflow/ewoksapps/ewoksid22/-/blob/id22_online_jul/src/ewoksid22/rebin.py

Reply:
If you call things from python, you don't have an `argparse` namespace object. Of course I could convert a dict to something that looks like a namespace object. However with the namespace object the parameter defaults are not defined by the `rebin_result_generator` function (unless inside you check the namespace for None's).

Both can work I suppose so it's a matter of what feels more natural.

This is still a draft. More testing today. It might change.